alluxio/worker_ring.py

```diff
@@ -283,30 +294,60 @@ def _fetch_workers_and_update_ring(self):
                     f"Failed to achieve worker info list from ETCD servers:{self._etcd_hosts}"
                 )
 
-        if worker_addresses != self._worker_addresses:
-            self._update_hash_ring(worker_addresses)
+        worker_info_map = {}
+        diff_in_worker_info_detected = False
+        for worker_entity in worker_entities:
+            worker_info_map[
+                worker_entity.worker_identity
+            ] = worker_entity.worker_net_address
+            if worker_entity.worker_identity not in self._worker_info_map:
+                diff_in_worker_info_detected = True
```

Comment:
we can early break here if we detect diff.

Reply:
worker_info_map[
                worker_entity.worker_identity
            ] = worker_entity.worker_net_address
            
 here we also need to update the local worker_info_map for update hash ring